### ISSUE TITLE
Regression in dispatcher/finalize thread

### ISSUE THREAD

**Issue by dginev** [dginev/CorTeX#42]
For some reason the finalize thread hits uniqueness violations over arxmliv, when running on supposedly new tasks for 02.2019. 
 * First: need to diagnose and resolve the uniqueness conflict
 * Second: this failure was too silent, panics in the finalize thread should panic the entire dispatcher process.

```
-- mark_done attempt failed: DatabaseError(UniqueViolation, "duplicate key value violates unique constraint \"log_fatals_task_id\"")
sink 164: incoming result for "tex_to_html", worker "monster.informatik.uni-erlangen.de:56", taskid: 9640054
sink 164: message size: 70819, took 2ms.
vent 187: worker "monster.informatik.uni-erlangen.de:56" received task 9640047
vent 187: message size: 6611612, took 229ms.
vent 188: worker "science-pup:17" received task 9640046
vent 188: message size: 5803616, took 127ms.
-- mark_done retry failed: DatabaseError(UniqueViolation, "duplicate key value violates unique constraint \"log_fatals_task_id\"")
sink 165: incoming result for "tex_to_html", worker "science-pup:17", taskid: 9641258
sink 165: message size: 488599, took 4ms.
-- mark_done retry failed: DatabaseError(UniqueViolation, "duplicate key value violates unique constraint \"log_fatals_task_id\"")
vent 189: worker "science-pup:08" received task 9640045
vent 189: message size: 1097618, took 57ms.
-- mark_done retry failed: DatabaseError(UniqueViolation, "duplicate key value violates unique constraint \"log_fatals_task_id\"")
thread '<unnamed>' panicked at 'Failed in finalize thread: StringError("Database ran away during mark_done persisting.")', src/dispatcher/manager.rs:95:27
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:39
   1: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at src/libstd/sys_common/backtrace.rs:59
             at src/libstd/panicking.rs:197
   3: std::panicking::default_hook
             at src/libstd/panicking.rs:211
   4: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:474
   5: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:381
   6: std::panicking::begin_panic_fmt
             at src/libstd/panicking.rs:336
   7: cortex::dispatcher::manager::TaskManager::start::{{closure}}::{{closure}}
```

**Comment by dginev** [comment]
My best guess, looking at the index that raised the error, is that a job returned with more than one Fatal message. So, two ways forward, either discard one Fatal message, or just allow multiple ones. I think allowing multiples can be more transparent in catching issues, so I will refactor in that direction.

**Comment by dginev** [comment]
Ah ok, this is pretty easy to fix. One is the more permissive index, a second part is to use a [poisoned mutex](https://doc.rust-lang.org/std/sync/struct.Mutex.html#poisoning) for the `done_queue`, so that panicking the finalize thread propagates to the ventilator thread, which is the first to join. A bit iffy to look at when it's done, but not the worst of patches.

**Comment by dginev** [comment]
Solved via d83768bda1f182a3ae9025b163127a3475a5aeb0 and 158ad18975f97af7a9664c86108a9abbb70895c4
